packages/next/src/server/app-render/metadata-insertion/create-server-inserted-metadata.tsx

```diff
@@ -35,19 +32,14 @@ export function createServerInsertedMetadata(nonce: string | undefined) {
     },
 
     async getServerInsertedMetadata(): Promise<string> {
-      if (!metadataResolver || metadataToFlush) {
+      if (inserted) {
         return ''
       }
 
-      metadataToFlush = metadataResolver()
+      inserted = true
       const html = await renderToString({
```

Comment:
This is such a simple script we should just use a string rather than render with React